html/pages/assistant.html

```diff
@@ -0,0 +1,245 @@
+					<div class="text-wrap" style="font-size: 14px;">Context Length: <span :class="getContextColor(contextLength)">{{ formatCount(contextLength) }}</span></div>
+				</v-card-title>
+				<v-divider></v-divider>
+				<v-card-text class="chat-messages pa-0">
+					<div class="pa-4">
+						<div v-for="(message, index) in messages" :key="index" class="message-item mb-3">
+							<div class="d-flex align-start">
+								<v-avatar size="32" class="mr-3" :color="message.role === 'user' ? 'primary' : 'secondary'">
+									<v-icon v-if="message.role === 'user'" class="text-white" size="small">fa-user</v-icon>
+									<img v-else src="images/favicon/onion.png" alt="Assistant" style="width: 24px; height: 24px;">
+								</v-avatar>
+								<div class="flex-grow-1">
+									<div class="d-flex align-center mb-1">
+										<span class="font-weight-medium text-body-2 mr-2">{{ message.role === 'user' ? 'You' : i18n.chat }}</span>
+										<span class="text-caption text-medium-emphasis mr-2">{{ formatTimestamp(message.timestamp) }}</span>
+										<span class="text-caption text-primary" v-if="message.usage && message.usage.credits">{{ message.usage.credits }} credits</span>
+									</div>
+									<div class="text-body-2" :key="message.content" v-html="formatMarkdown(message.content)"></div>
+									
+									<!-- Tool Use Display -->
+									<div v-if="message.toolUses && message.toolUses.length > 0" class="mt-3">
+										<div v-for="toolUse in message.toolUses" :key="toolUse.id" class="tool-use-card mb-2">
+											<v-card variant="outlined" class="pa-3">
+												<div class="d-flex align-center mb-2">
+													<v-icon :color="getToolStatusColor(toolUse.status)" class="mr-2" size="small">
+														{{ getToolStatusIcon(toolUse.status) }}
+													</v-icon>
+													<span class="font-weight-medium text-body-2">{{ toolUse.name }}</span>
+													<v-spacer></v-spacer>
+													<v-chip :color="getToolStatusColor(toolUse.status)" size="x-small" variant="flat">
+														{{ toolUse.status }}
+													</v-chip>
+												</div>
+												
+												<!-- Tool Parameters -->
+												<div v-if="Object.keys(toolUse.input).length > 0" class="mb-2">
+													<div class="text-caption text-medium-emphasis mb-1">Parameters:</div>
+													<v-code class="text-caption">{{ JSON.stringify(toolUse.input, null, 2) }}</v-code>
+												</div>
+												
+												<!-- Tool Approval Buttons -->
+												<div v-if="toolUse.status === 'pending_approval'" class="mb-2">
+													<div class="text-caption text-medium-emphasis mb-1">This tool wants to execute. Do you approve?</div>
+													<div class="d-flex tool-approval-buttons">
+														<v-btn
+															color="success"
+															size="small"
+															variant="flat"
+															@click="approveTool(toolUse)"
+															data-aid="approve_tool"
```

Comment:
data aid format: uicomponents_uniquename_type (type is optional and you can see when I've used it and when I haven't by looking at existing patterns). All data-aids need to be reviewed and made to follow the existing pattern.